cumulus/client/cirrus-executor/src/processor.rs

```diff
@@ -136,11 +136,13 @@ where
 
 		tracing::trace!(target: LOG_TARGET, ?extrinsics, "Origin deduplicated extrinsics");
 
-		let block_number = self.client.info().best_number;
+		let parent_hash = self.client.info().best_hash;
+		let parent_number = self.client.info().best_number;
+
 		let extrinsics: Vec<_> = match self
 			.runtime_api
 			.runtime_api()
-			.extract_signer(&BlockId::Number(block_number), extrinsics)
+			.extract_signer(&BlockId::Number(parent_number), extrinsics)
```

Comment:
Let's use hash instead of number here, I can imagine block number is more expensive and there could be edge cases during forking, hash is safer. Here and below.